backend/src/main/kotlin/org/loculus/backend/service/submission/CurrentProcessingPipelineTable.kt

```diff
@@ -1,11 +1,53 @@
 package org.loculus.backend.service.submission
 
+import kotlinx.datetime.LocalDateTime
 import org.jetbrains.exposed.sql.Table
+import org.jetbrains.exposed.sql.andWhere
+import org.jetbrains.exposed.sql.batchInsert
 import org.jetbrains.exposed.sql.kotlin.datetime.datetime
+import org.jetbrains.exposed.sql.selectAll
+import org.jetbrains.exposed.sql.update
 
 const val CURRENT_PROCESSING_PIPELINE_TABLE_NAME = "current_processing_pipeline"
 
 object CurrentProcessingPipelineTable : Table(CURRENT_PROCESSING_PIPELINE_TABLE_NAME) {
+    val organismColumn = varchar("organism", 255)
     val versionColumn = long("version")
     val startedUsingAtColumn = datetime("started_using_at")
+
+    /**
+     * Every organism needs to have a current pipeline version in the CurrentProcessingPipelineTable.
+     * This function sets V1 for all given organisms, if no version is defined yet.
+     */
+    fun setV1ForOrganismsIfNotExist(organisms: Collection<String>, now: LocalDateTime) =
+        CurrentProcessingPipelineTable.batchInsert(organisms, ignore = true) { organism ->
+            this[organismColumn] = organism
+            this[versionColumn] = 1
+            this[startedUsingAtColumn] = now
+        }
+
+    /**
+     * Given a version that was found, is the currently stored 'current' pipeline version for this organism
+     * less than the one that was found? If so, the pipeline needs to 'update' i.e. reprocess older entries.
+     */
+    fun pipelineNeedsUpdate(foundVersion: Long, organism: String) = CurrentProcessingPipelineTable
+        .selectAll()
+        .where { versionColumn less foundVersion }
+        .andWhere { organismColumn eq organism }
+        .limit(1)
```

Comment:
The limit is not really needed, right? Since it should be at most one?

Reply:
I based this on previous code where the same applied, I thought maybe it was there to make the query faster or something. But I can remove it

Reply:
https://github.com/loculus-project/loculus/pull/3534/commits/9c8e7cac8bc748c5a2c544339d0d522262f66d8c